src/gsd.ts

```diff
@@ -111,42 +111,35 @@ export function gsd(data: DataXY, options: GSDOptions = {}): GSDPeakID[] {
 
   let yData = y;
   let dY, ddY;
-  const { windowSize, polynomial } = sgOptions;
 
   if (equallySpaced) {
     if (smoothY) {
       yData = sgg(y, x[1] - x[0], {
-        windowSize,
-        polynomial,
+        ...sgOptions,
```

Comment:
We send other parameters that just windowSize and polynomial ?

If sgg does not change the options (like adding a new parameter) and this would be a bug in sgg I think, you don't need to create a new object with destructuring sgOptions. You can directly send sgOptions.